nil pointer in (*Association).AcceptStream

### Your environment.
  - Version: 1.4.2

### What did you do?
I pushed to github and the CI ran the tests on my rtpdump [branch](https://github.com/pions/webrtc/pull/522).

### What happened?

```
=== RUN   TestDataChannel_ORTCE2E
Failed to read from data channel association closed
Failed to read from data channel association closed
Failed to accept data channel: The association is closed
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7511fe]
goroutine 92 [running]:
github.com/pions/sctp.(*Association).AcceptStream(...)
	/home/travis/gopath/pkg/mod/github.com/pions/sctp@v1.4.2/association.go:579
github.com/pions/datachannel.Accept(0x0, 0x1, 0xc000000006, 0x28)
	/home/travis/gopath/pkg/mod/github.com/pions/datachannel@v1.2.0/datachannel.go:73 +0x3e
github.com/pions/webrtc.(*SCTPTransport).acceptDataChannels(0xc0000a64e0)
	/home/travis/build/pions/webrtc/sctptransport.go:133 +0x10d
created by github.com/pions/webrtc.(*SCTPTransport).Start
	/home/travis/build/pions/webrtc/sctptransport.go:96 +0x1a4
FAIL	github.com/pions/webrtc	7.619s
```

Here are some other errors looking through `travis-ci` history that might be a clue

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7f7b1e]
goroutine 49 [running]:
github.com/pion/sctp.(*Association).AcceptStream(...)
	/home/travis/gopath/pkg/mod/github.com/pion/sctp@v1.6.4/association.go:1017
github.com/pion/datachannel.Accept(0x0, 0xc00046ffa0, 0xc000000003, 0x47547b, 0xc0002ca468)
	/home/travis/gopath/pkg/mod/github.com/pion/datachannel@v1.4.5/datachannel.go:126 +0x3e
github.com/pion/webrtc/v2.(*SCTPTransport).acceptDataChannels(0xc000468700)
	/home/travis/build/pion/webrtc/sctptransport.go:134 +0x18f
created by github.com/pion/webrtc/v2.(*SCTPTransport).Start
	/home/travis/build/pion/webrtc/sctptransport.go:97 +0x232
FAIL	github.com/pion/webrtc/v2	2.236s
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7f7a4e]
goroutine 980 [running]:
github.com/pion/sctp.(*Association).AcceptStream(...)
	/home/travis/gopath/pkg/mod/github.com/pion/sctp@v1.6.4/association.go:1017
github.com/pion/datachannel.Accept(0x0, 0xc000474fa0, 0xc000000033, 0x47547b, 0xc000339a98)
	/home/travis/gopath/pkg/mod/github.com/pion/datachannel@v1.4.5/datachannel.go:126 +0x3e
github.com/pion/webrtc/v2.(*SCTPTransport).acceptDataChannels(0xc00036a620)
	/home/travis/build/pion/webrtc/sctptransport.go:134 +0x18f
created by github.com/pion/webrtc/v2.(*SCTPTransport).Start
	/home/travis/build/pion/webrtc/sctptransport.go:97 +0x232
FAIL	github.com/pion/webrtc/v2	98.611s
```

--- a/sctptransport.go
+++ b/sctptransport.go
@@ -94,7 +94,7 @@ func (r *SCTPTransport) Start(remoteCaps SCTPCapabilities) error {
 	}
 	r.association = sctpAssociation
 
-	go r.acceptDataChannels()
+	go r.acceptDataChannels(sctpAssociation)
 
 	return nil
 }
@@ -126,10 +126,7 @@ func (r *SCTPTransport) ensureDTLS() error {
 	return nil
 }
 
-func (r *SCTPTransport) acceptDataChannels() {
-	r.lock.RLock()
-	a := r.association
-	r.lock.RUnlock()
+func (r *SCTPTransport) acceptDataChannels(a *sctp.Association) {
 	for {
 		dc, err := datachannel.Accept(a, &datachannel.Config{
 			LoggerFactory: r.api.settingEngine.LoggerFactory,

